python/lsst/cp/pipe/cpFunctions.py

```diff
@@ -124,6 +115,12 @@ def registerCalibrations(self, datasetTypeName):
         self.butler.collection = None
 
         for newId, data in zip(self.newDataId, self.objects):
+            # Special case known special storageClasses.
+            if datasetTypeName in ('bias', 'dark'):
+                data = data.getImage()
+            elif datasetTypeName in ('flat'):
+                data = data.getMaskedImage()
+
```

Comment:
If this code should be there, put it there in the original.

Reply:
Unexpected type mismatches between HSC (special cased when I made the gen3 repo two months ago) and LATISS (no special cases, so must properly type the data).